File: website/src/api/repositories/order.ts
import { client } from "../../utils/aspida";
import {
  GetOrderResponseInterface,
  GetOrdersResponseInterface,
} from "../interfaces/order";

export const getOrders = async (userId?: string) => {
  try {
    const res: GetOrdersResponseInterface = await client.v1.orders.$get({
      query: { userId },
    });
    return res;
  } catch (err) {
    console.log(err);
  }
};

export const getOrder = async (orderId: string) => {
  try {
    const res: GetOrderResponseInterface = await client.v1.orders
      ._orderId(orderId)
      .$get();
    return res;
  } catch (err) {
    console.log(err);
  }
};

File: backend/internal/adapter/gateway/repository/order.go
package repository

import (
	"context"

	"github.com/hal-cinema-2024/backend/internal/entities/model"
	"github.com/hal-cinema-2024/backend/internal/framework/herror"
	"github.com/hal-cinema-2024/backend/internal/usecase/dai"
	"gorm.io/gorm"
)

type OrderRepo struct {
	db *gorm.DB
}

func NewOrderRepo(db *gorm.DB) *OrderRepo {
	return &OrderRepo{db: db}
}

func (r *OrderRepo) CreateOrder(ctx context.Context, order *model.Order) error {
	return r.db.Create(order).Error
}

func (r *OrderRepo) GetOrder(ctx context.Context, orderID string) (*model.Order, error) {
	var order model.Order
	if err := r.db.Where("id = ?", orderID).First(&order).Error; err != nil {
		return nil, err
	}
	return &order, nil
}

func (r *OrderRepo) GetOrdersByUserID(ctx context.Context, userID string) ([]*model.Order, error) {
	var orders []*model.Order
	if err := r.db.Where("user_id = ?", userID).Find(&orders).Error; err != nil {
		return nil, err
	}
	return orders, nil
}

func (r *OrderRepo) UpdateOrder(ctx context.Context, order *model.Order) error {
	if order.OrderID == "" {
		return herror.ErrRequired
	}

	if order.UserID == "" {
		return herror.ErrRequired
	}

	return r.db.Save(order).Error
}

func (r *OrderRepo) DeleteOrder(ctx context.Context, orderID string) error {
	return r.db.Delete(&model.Order{}, orderID).Error
}

var _ dai.OrderRepo = (*OrderRepo)(nil)
